module/VuFind/src/VuFind/Db/Table/Resource.php

```diff
@@ -246,11 +244,16 @@ public function updateRecordId($oldId, $newId, $source = DEFAULT_SEARCH_BACKEND)
             if ($newResource = $this->findResource($newId, $source, false)) {
                 // Special case: merge new ID and old ID:
                 foreach (['comments', 'userresource', 'resourcetags'] as $table) {
-                    $tableObjects[$table] = $this->getDbTable($table);
-                    $tableObjects[$table]->update(
-                        ['resource_id' => $newResource->id],
-                        ['resource_id' => $resource->id]
-                    );
+                    if ($table == 'resourcetags') {
+                        $tableObjects[$table] = $this->getDbService(\VuFind\Db\Service\TagService::class);
+                        $tableObjects[$table]->update($newResource->id, $resource->id);
```

Comment:
Since we're breaking the pattern of the loop, maybe it's better to do the database service stuff outside of the loop instead of introducing a special case here. It also looks like there is no `update` method on the tag service -- I think you mean to be calling `updateResource` but maybe a better name for that method would be `changeResourceId`.

Reply:
It was left as tableObjects, after this I will migrate rest of the things and change it to serviceObjects under a loop.